include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -397,14 +398,12 @@ namespace internal
           return;
         }
       else if (quad_in.is_tensor_product() == false ||
-               dynamic_cast<const FE_SimplexP<dim> *>(
-                 &fe_in.base_element(base_element_number)) ||
-               dynamic_cast<const FE_SimplexDGP<dim> *>(
-                 &fe_in.base_element(base_element_number)) ||
-               dynamic_cast<const FE_WedgeP<dim> *>(
-                 &fe_in.base_element(base_element_number)) ||
-               dynamic_cast<const FE_PyramidP<dim> *>(
-                 &fe_in.base_element(base_element_number)))
+               dynamic_cast<const FE_SimplexPoly<dim, dim> *>(
+                 &fe_in.base_element(base_element_number)) != nullptr ||
+               dynamic_cast<const FE_WedgePoly<dim, dim> *>(
+                 &fe_in.base_element(base_element_number)) != nullptr ||
+               dynamic_cast<const FE_PyramidPoly<dim, dim> *>(
+                 &fe_in.base_element(base_element_number)) != nullptr)
```

Comment:
This was not correct and not an issue, since the first condition normally is true.